.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: SROS2 CI
+on:
+  pull_request:
+  push:
+  schedule:
+    # Run daily
+    - cron:  '0 20 * * *'
+
+jobs:
+  test_latest:
+    runs-on: ubuntu-latest
+    container: osrf/ros2:devel
+    steps:
+    - run: |
+        apt-get -qq update
+        apt-get -qq upgrade -y
+        apt-get -qq install -y curl libasio-dev libtinyxml2-dev
+    - uses: mikaelarguedas/action-ros-ci@sros2-version
+      with:
+        package-name: |
+          sros2
+          sros2_cmake
+        # skipping end-to-end tests for now https://github.com/eProsima/Fast-RTPS/issues/1087
+        #  test_security
+        # extra-cmake-args: '-DSECURITY=ON --no-warn-unused-cli'
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1.0.6
+      if: github.event_name == 'pull_request' || (github.event_name == 'push' && github.ref == 'refs/heads/master')
+      with:
+        file: ros_ws/build/sros2/coverage.xml
+        flags: unittests
+        name: sros2-coverage
+        fail_ci_if_error: true
+        yml: .codecov.yml
```

Comment:
Do we have the ability to fail CI if coverage decreases?

Reply:
Not at the moment. And it seems hard to implement outside of the codecov machinery as the job itself doesn't know past coverage, only the coverage status of the current run. Past values are stored on codecov.io and the comparison and feedback is handled on their side.
However codecov will post a comment on the PR with the coverage diff (e.g. https://github.com/mikaelarguedas/sros2/pull/2#issuecomment-604699896) and that will give both contributor and maintainer notification of coverage going up or down on the PR.

Reply:
Alright fair enough.

Reply:
To clarify this a bit further, the github actions job will not fail, but codecov does post a comment *and* status checks on the PR (codecov/project and codecov/patch). If the patch decreases coverage, the codecov status check will be a failed one. Valid example at the time of the writing https://github.com/ros-infrastructure/rosdep/pull/712